recipes/mimalloc/all/conanfile.py

```diff
@@ -0,0 +1,123 @@
+            self._cmake_i.definitions["MI_BUILD_TESTS"] = False
+            if self.options.shared:
+                self._cmake_i.definitions["MI_BUILD_SHARED"] = True
+                self._cmake_i.definitions["MI_BUILD_STATIC"] = False
+            else:
+                self._cmake_i.definitions["MI_BUILD_SHARED"] = False
+                self._cmake_i.definitions["MI_BUILD_STATIC"] = True
+            self._cmake_i.configure(source_folder=self._source_subfolder,
+                build_folder=self._build_subfolder)
+        return self._cmake_i
+
+    @property
+    def _msbuild(self):
+        if not self._msbuild_i:
+            self._msbuild_i = MSBuild(self)
+        return self._msbuild_i
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("mimalloc-" + self.version, self._source_subfolder)
+
+    def requirements(self):
+        pass
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        tools.check_min_cppstd(self, "14")
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.compiler == "Visual Studio":
+            if Version(self.settings.compiler.version) < "15":
+                raise ConanInvalidConfiguration("Visual Studio 15 2017 or newer is required")
+
+    def build(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+
+        if self.settings.compiler == "Visual Studio":
+            target = "mimalloc-override" if self.options.shared else "mimalloc"
+            if Version(self.settings.compiler.version) >= "16":
+                self._msbuild.build(
+                    os.path.join(self._source_subfolder, "ide", "vs2019", "mimalloc.sln"),
+                    targets=[target])
+            else:
+                self._msbuild.build(
+                    os.path.join(self._source_subfolder, "ide", "vs2017", "mimalloc.sln"),
+                    targets=[target])
+        else:
+            self._cmake.build()
+
+    def package(self):
+        if self.settings.compiler == "Visual Studio":
+            self.copy("*.h", dst="include",
+                src=os.path.join(self._source_subfolder, "include"))
+            self.copy("*.lib", dst="lib",
+                src=os.path.join(self._source_subfolder, "out"),
+                keep_path=False)
+            self.copy("*.dll", dst="bin",
+                src=os.path.join(self._source_subfolder, "out"),
+                keep_path=False)
+        else:
+            self._cmake.install()
+            tools.rmdir(os.path.join(self.package_folder, "lib", self._cmake_install_folder, "cmake"))
+
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
+        if self.settings.compiler != "Visual Studio":
```

Comment:
I think it also needs some system libraries:
https://github.com/microsoft/mimalloc/blob/a9686d6ecf00e4467e772f7c0b4ef76a15f325f6/CMakeLists.txt#L193-L199

Reply:
Maybe this will suffice: a2901ec ?

Btw, do you know how the rt library is expected to be dealt with? I added it as a system library for Linux builds...

Reply:
- I would add these system libraries only if the library is built as a static library. As shared libraries should not expose this dependency.
- What the other recipes do, is add pthread and rt only for Linux.

Reply:
Done, cf. c5a1db46